apps/web/src/app/(onboarding)/onboarding-devs/page.tsx

```diff
@@ -0,0 +1,41 @@
+import { redirect } from 'next/navigation'
+import { isOnboardingCompleted } from '$/data-access/workspaceOnboarding'
+import { getCurrentUserOrRedirect } from '$/services/auth/getCurrentUser'
+import { NotFoundError } from '@latitude-data/core/lib/errors'
+import { OnboardingClient } from './_components/OnboardingClient'
+import { findOnboardingDocument } from '@latitude-data/core/services/documents/findOnboardingDocument'
+import { findOnboardingDataset } from '@latitude-data/core/services/datasets/findOnboardingDataset'
+
+export default async function OnboardingRedirect() {
+  const isCompleted = await isOnboardingCompleted()
+  if (isCompleted) {
+    redirect('/dashboard')
+  }
+
+  const { workspace } = await getCurrentUserOrRedirect()
+  if (!workspace?.id) {
+    throw new NotFoundError('Workspace ID is required')
```

Comment:
Use [notFound](https://nextjs.org/docs/app/api-reference/functions/not-found) from next.js